File: diplomski/src/main/java/com/miocinovic/rastko/diplomski/service/UserService.java
package com.miocinovic.rastko.diplomski.service;

import java.util.List;

import com.miocinovic.rastko.diplomski.dto.UserDetailsDto;
import com.miocinovic.rastko.diplomski.model.User;

public interface UserService {
	public User saveUser(User user);

	public UserDetailsDto getUserDetails(String username);

	public void setUserDetails(String username, UserDetailsDto detail);

	public void addContact(String userA, String userB);

	public List<UserDetailsDto> getContacts(String username);

	List<UserDetailsDto> getUserDetailsByLocation(String location);
}

File: web/diplomskifrontend/src/services/user.service.js
import axios from 'axios';
import authHeader from './auth-header';
import authService from './auth.service';

const USER_API_URL = 'http://localhost:8080/user/';

class UserService {
    getUserDetails(username) {
        return axios.get(USER_API_URL + username, { headers: authHeader() });
    }

    updateUserDetails(details) {
        return axios.put(USER_API_URL + authService.getCurrentUser().username, details, { headers: authHeader() });
    }

    getUserDetailsByLocation(location) {
        return axios.get(USER_API_URL + 'location/' + location, { headers: authHeader() });
    }

    getContacts() {
        return axios.get(USER_API_URL + authService.getCurrentUser().username + '/contacts', { headers: authHeader() });
    }

    addContact(user) {
        return axios.put(USER_API_URL + authService.getCurrentUser().username + '/contacts', user, { headers: authHeader() });
    }
}

export default new UserService();
